lib/chef/provider/aws_subnet.rb

```diff
@@ -6,22 +6,40 @@ class Chef::Provider::AwsSubnet < Chef::Provider::AwsProvider
   action :create do
     fail "Can't create a Subnet without a CIDR block" if new_resource.cidr_block.nil?
 
-    if existing_subnet == nil
-      converge_by "Creating new Subnet #{fqn} in VPC #{new_resource.vpc} in #{new_resource.region_name}" do
+    if existing_subnet
+      subnet = existing_subnet
+      # TODO update things
+    else
+      converge_by "Creating new Subnet #{fqn} in VPC #{new_resource.vpc} in #{new_driver.aws_config.region}" do
         opts = { :vpc => vpc_id }
         opts[:availability_zone] = new_resource.availability_zone if new_resource.availability_zone
-        subnet = ec2.subnets.create(new_resource.cidr_block, opts)
+        subnet = new_driver.ec2.subnets.create(new_resource.cidr_block, opts)
         subnet.tags['Name'] = new_resource.name
         subnet.tags['VPC'] = new_resource.vpc
         new_resource.subnet_id subnet.id
         new_resource.save
       end
     end
+
+    subnet_desc = new_driver.ec2.client.describe_subnets(subnet_ids: [ subnet.id ])[:subnet_set].first
+    if new_resource.map_public_ip_on_launch
+      if !subnet_desc[:map_public_ip_on_launch]
+        converge_by "Turning on automatic public IPs for subnet #{subnet.id}" do
+          new_driver.ec2.client.modify_subnet_attribute(subnet_id: subnet.id, map_public_ip_on_launch: { value: true })
```

Comment:
Can't you set this attribute during the create?  http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/EC2/Client.html#create_subnet-instance_method says you can